runtime/common/src/assigned_slots.rs

```diff
@@ -0,0 +1,1086 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! This pallet allows to assign permanent (long-lived) or temporary
+//! (short-lived) parachain slots to paras, leveraging the existing
```

Comment:
nitpick: the permanent leases are not permanent, but just long-lived :stuck_out_tongue:

Reply:
Fair enough .. they're permanent until they're unassigned (and for as long the `PermanentSlotLeasePeriodLength` lasts). It's just less of a mouthful than long-lived (or short-lived for temporary).